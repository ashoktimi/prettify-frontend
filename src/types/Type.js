import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import PrettifyApi from "../api/api";
import LoadingSpinner from "../helpers/LoadingSpinner";
import ProductCard from "../products/productCard";
import NavbarBottom from '../routesNav/NavBottom';
import { Row, Col } from 'react-bootstrap';

const Type = () =>{
    const { id, type } = useParams();
    const [data, setData] = useState ({});

    useEffect(() => {
        async function getData (id){
              let response = await PrettifyApi.getTypeDetail(id, type);
              setData(response);            
        }
        getData(id)
      }, [id, type]);

      if (!data) return <LoadingSpinner />;

      return(
        <>
         {data.products && data.products.length >0 ?
           <Row style={{  width: '80%', left:0, right:0, margin: 'auto', marginTop: '2rem' }}> 
            <div>
              <h3 className="Head-Name">{(data.name).trim().replace("_", " ")}</h3>   
            </div>     
            {data.products.map(p => (
              <Col key={p.id} xs={6} md={4}  style={{ height: '500px'}}>  
                <ProductCard 
                  id={p.id}
                  name={p.name} 
                  price={p.price}
                  priceSign={p.price_sign}
                  prevPrice={p.prev_price}
                  image_link={p.image_link}
                  rating={p.rating}
                  numOfRating={p.number_rating}
                />  
              </Col>       
            ))}
           </Row>
          :
          null 
          }
        <NavbarBottom/>
      </>
    )
  };

export default  Type;