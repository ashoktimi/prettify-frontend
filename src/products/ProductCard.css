
.ProductCard{
  position: relative;
  z-index: 1;
  height: 95%;
  width: 100%;
  background-color: white;
  display: flex;
  flex-direction: column;
}
.ProductCard-imgDiv{
  height: 300px;
  overflow: hidden;
  background-color: rgb(219, 204, 205);
}
.img{
  object-fit: cover;
  height: 100%;
  width: 100%;
  padding-top: 2%;
  background-color: rgb(255, 252, 252);
}

.ProductCard-body{
  height: 150px;
  border-radius: .5rem;
  overflow: hidden;
  background-color: white;
  width: 80%;
  left: 0;
  right: 0;
  margin: auto;
  position: relative;
}

.productAddBtn{
  color: black;
  position: absolute; 
  bottom: 12px;
  width: 95%;
  background-color: rgb(37, 150, 190);
}

.ProductCard-price {
font-size: 15px;
margin: 0;
color: black;
display: inline-block;  
font-weight: bold;
}
.priceBefore{
text-decoration: line-through;
}
.ProductCard-rating{
display: flex;
justify-content: space-between;
}
.ProductCard-star {
margin-right: 0.5rem; 
}
.Product-count{
display: inline-block;
}