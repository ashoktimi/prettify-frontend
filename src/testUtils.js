import React from "react";
import UserContext from "./auth/userContext";


const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, handleToggleOffcanvas = () => false }) => (
    <UserContext.Provider value={{ currentUser, handleToggleOffcanvas }}>
      {children}
    </UserContext.Provider>
);

function setup(){
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  });
}

function teardown(){
   window.matchMedia.mockClear();
   delete window.matchMedia;
}

function commonBeforeEach() {
  setup();
}

function commonAfterEach() {
  teardown();
}

export { UserProvider, commonBeforeEach, commonAfterEach };