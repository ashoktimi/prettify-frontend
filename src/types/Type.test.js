import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { UserProvider,  commonBeforeEach, commonAfterEach } from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from '../cart/cartSlice';
import Type from './Type';
import ProductCard from "../products/productCard";
import LoadingSpinner from "../helpers/LoadingSpinner";
import NavbarBottom from '../routesNav/NavBottom';

beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('Type', () => {
  const store = configureStore({
    reducer: {
      cart: cartReducer
    }
  });
test("renders without crashing", function () {
  render(
   <UserProvider>
     <Type />
   </UserProvider>    
   );
});

test("matches snapshot", function () {
  const { asFragment } = render(<Type />);
  expect(asFragment()).toMatchSnapshot();
});

test("ProductCard component renders correctly", function () {
  render(
    <MemoryRouter>
     <Provider store={store}>
      <UserProvider>
        <ProductCard /> 
      </UserProvider> 
     </Provider>     
    </MemoryRouter>   
   );
});

test("LoadingSpinner component renders correctly ", function () {
  render(
    <MemoryRouter>
       <LoadingSpinner />
    </MemoryRouter>   
   );
});

test("NavbarBottom component renders correctly ", function () {
  render(
    <MemoryRouter>
       <NavbarBottom />
    </MemoryRouter>   
   );
});
})


