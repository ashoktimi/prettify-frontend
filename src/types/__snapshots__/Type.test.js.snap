// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</DocumentFragment>
`;
